parser.cpp

```diff
@@ -238,6 +236,26 @@ class Parser {
         return token::IDENTIFIER;
     }
 
+    Token parse_number_literal(const char** data) {
+        auto str = *data;
+        while (is_not_at_eof()) {
+            auto c = *str;
+            if (isdigit(c)) {
+                str++;
+                pos++;
+            } else if (c == '(' || c == ',' || c == ')' || c == '.') {
```

Comment:
I think a check for whitespace is missing here.

Reply:
Also does this set the next position right before any of these characters? (For correct positions of later tokens?)

Reply:
Yes, when it comes out of the function, `pos` will be pointing to one of these characters. Or to put it another way, `pos` in only incremented for a digit.